samples/drivers/CAN/src/main.c

```diff
@@ -199,6 +199,10 @@ void main(void)
 
 #ifdef CONFIG_LOOPBACK_MODE
```

Comment:
In my opinion we can change this in general an not only for loopback. Can you change the data rate in device tree?